server/events/vcs/gitlab_client_test.go

```diff
@@ -323,8 +323,8 @@ func TestGitlabClient_PullIsMergeable(t *testing.T) {
 	gitlabVersionUnder15_6 := "15.3.2-ce"
 	gitlabServerVersions := []string{gitlabVersionOver15_6, gitlabVersion15_6, gitlabVersionUnder15_6}
 	vcsStatusName := "atlantis-test"
-	defaultMR = 1
-	noHeadPipelineMR = 2
+	defaultMr := 1
+	noHeadPipelineMR := 2
```

Comment:
@marceloboeira apparently that wasn't enough, tests are still failing:
```
=== RUN   TestGitlabClient_PullIsMergeable/atlantis-test/plan#02
    gitlab_client_test.go:423: got unexpected request at "/api/v4/projects/4580910/repository/commits/sha/statuses"
    gitlab_client_test.go:451: unexpected error: GET http://127.0.0.1:52736/api/v4/projects/4580910/repository/commits/sha/statuses: 404 failed to parse unknown error format: not found
        
    --- FAIL: TestGitlabClient_PullIsMergeable/atlantis-test/plan#02 (1.09s)
```

Is a case missing in https://github.com/runatlantis/atlantis/blob/4b340552dc3a0f017f8fccccc99d1cdb8897f287/server/events/vcs/gitlab_client_test.go#L397-L425?

Reply:
Ohh 😮‍💨 

thanks for bringing this up.

I'm curious how the tests passed din the MR build... let me take a look

Reply:
They were skipped 😕 https://github.com/runatlantis/atlantis/pull/3653#issuecomment-1666815423

Reply:
Could you try changing like 448 from "sha" to "67cb91d3f6198189f433c045154a885784ba6977"?

https://github.com/runatlantis/atlantis/pull/3653/files#diff-0cf0d1deb8569e72ff771612d80c7f49b7c947e8d2bf62b50562cf6cae54b493R448

Reply:
I think that's what's causing the problem since the mock is expecting a commit hash instead of "sha" in the URL. Alternatively we could create a mock for "sha" for isolation but I don't see an additional value.

Reply:
@marceloboeira that solves most of the failures, but still some test cases fail with:
```
    gitlab_client_test.go:452: [true != false]
```

This test case:
https://github.com/runatlantis/atlantis/blob/63dcd436b00f1183c4c2e175f8cd41185f906b78/server/events/vcs/gitlab_client_test.go#L384-L390

Reply:
I'm wondering if we have to revert the MR so we fix the build, I won't have time to look into it, it might be the default to `isPipelineSkipped := true` is the problem but I will have to open another MR with the fix (and figure out if we can have the CI running the tests or get them working properly locally)

Reply:
@marceloboeira that is indeed the case. I don't have much experience with Gitlab API so I can't tell what should be the correct behavior here.

Unless you think that the broken test might mean an incorrect behavior, how about removing this test case for now and then later you could look into it?

Reply:
The code definitely fixes the nil pointer (which caused panic) but by fixing it it might have exposed a logic issue. In any case, it is very niche. If you want to remove/comment the test I will open another MR soon to fix it

Reply:
Thanks, if you manage to merge your pull request first, I will close this one.